packages/frontend/src/features/semanticViewer/components/SaveChart/SemanticViewerSaveChartContent.tsx

```diff
@@ -28,7 +44,48 @@ const SemanticViewerSaveChartContent: FC = () => {
         dispatch(updateSaveModalOpen(true));
     };
 
-    const canSave = selectedFieldNames.length > 0 && !!selectedChartConfig;
+    const chartUpdateMutation = useSavedSemanticViewerChartUpdateMutation({
+        projectUuid,
+    });
+
+    const handleUpdate = useCallback(async () => {
+        if (
+            !savedSemanticViewerChartUuid ||
+            !activeChartKind ||
+            !selectedChartConfig
+        )
+            return;
+
+        await chartUpdateMutation.mutateAsync({
+            uuid: savedSemanticViewerChartUuid,
+            payload: {
+                versionedData: {
+                    chartKind: activeChartKind,
+                    config: selectedChartConfig,
+                    semanticLayerQuery,
+                    // TODO: view should never be ''. this is a temporary fix for semantic layers without views
+                    semanticLayerView: semanticLayerView ?? '',
+                },
+            },
+        });
+
+        showToastSuccess({
+            title: 'Chart saved successfully!',
+        });
+    }, [
+        activeChartKind,
+        chartUpdateMutation,
+        savedSemanticViewerChartUuid,
+        selectedChartConfig,
+        semanticLayerQuery,
+        semanticLayerView,
+        showToastSuccess,
+    ]);
+
+    const canSave =
+        selectedFieldNames.length > 0 &&
+        !!selectedChartConfig &&
+        !!activeChartKind;
```

Comment:
This is most likely the cause of the issue here: https://github.com/lightdash/lightdash/pull/11659#issuecomment-2374168083

We no longer require a activeChartKind to save, it should default to table

Reply:
ah, makes sense!